### ISSUE TITLE
False positive: suspicious_arithmetic_impl triggered by bitflags Sub

### ISSUE THREAD

**Issue by niklasf** [rust-lang-nursery/rust-clippy#2510]
`Sub` in bitflags is [implemented](https://github.com/rust-lang-nursery/bitflags/blob/13e54b43c92c9ef170b6a6a122da82d8f5c0d283/src/lib.rs#L641-L649) like so:

```rust
        impl $crate::_core::ops::Sub for $BitFlags {
            type Output = $BitFlags;


            /// Returns the set difference of the two sets of flags.
            #[inline]
            fn sub(self, other: $BitFlags) -> $BitFlags {
                $BitFlags { bits: self.bits & !other.bits }
            }
        }
```

This is correct but triggers `suspicious_arithmetic_impl`.

```
46 | / bitflags! {
...  |
60 | | }
   | |_^
   |
   = note: #[warn(suspicious_arithmetic_impl)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.187/index.html#suspicious_arithmetic_impl
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

**Comment by oli-obk** [comment]
cc @flip1995 

The bit-inversion should probably count as another operation and thus not trigger the lint.

Opinions? Maybe this is a special case that should simply `allow` that lint?

**Comment by flip1995** [comment]
Yeah the bit inversion should count as another operation. I will fix this asap.
